runtime/debugger.c

```diff
@@ -170,16 +169,38 @@ static void winsock_cleanup(void)
 {
   WSACleanup();
 }
+
+static bool is_implicit(const char *path)
+{
+  size_t len = strlen(path);
+  return (len < 1 || path[0] != '/')
+    && (len < 1 || path[0] != '\\')
+    && (len < 1 || path[1] == 0 || path[1] != ':')
+    && strncmp(path, "./", 2) != 0
+    && strncmp(path, ".\\", 2) != 0
+    && strncmp(path, "../", 3) != 0
+    && strncmp(path, "..\\", 3) != 0;
+}
+#else
+static bool is_implicit(const char *path)
+{
+  size_t len = strlen(path);
+  return (len < 1 || path[0] != '/')
+    && strncmp(path, "./", 2) != 0
+    && strncmp(path, "../", 3) != 0;
+}
 #endif
+static bool is_ipv6(const char *address, const char *port)
+{
+  return address[0] == '[' && port[-1] == ']';
```

Comment:
I am not really fond of `port[-1]` here - if you pass `":"` as the debugger socket specification, you will invoke this function with `port == address` and only skirt the out-of-bounds access because it will fail the `address[0] == '['` check.

Reply:
> only skirt the out-of-bounds access because it will fail the `address[0] == '['` check.
I was proud of this trick but I understand if you think that's not a great style.